Add 2FA paragraph (#479)
NethServer/docs: administrator-manual/en/base_system2.rst

* dashboard
* applications
* terminal

Two-factor authentication (2FA)
===============================

> Two-factor authentication (2FA) is an extra layer of security required to access the new Server Manager.

Reviewer: Two-factor authentication (2FA) is an (optional) extra layer of security...

Author: It's not optional once enforced.

> Two-factor authentication (2FA) is an extra layer of security required to access the new Server Manager.

Reviewer 2: ```suggestion
Two-factor authentication (2FA) is an extra layer of security that can be added to access the new Server Manager.
```

Author: I do not like it.

What about:
```
Two-factor authentication (2FA) adds extra layer of security required ro access the new Server Manager.
```

@filippocarletti ?

Reviewer 2: Pay attention to the "required ro" typo :)

I think Filippo's idea is to suggest that the 2FA is not mandatory. The extra layer is optional.

Reviewer: ```suggestion
Two-factor authentication (2FA) can be used to add an extra layer of security required to access the new Server Manager.
```

> Two-factor authentication (2FA) is an extra layer of security required to access the new Server Manager.

Reviewer: ```suggestion
Two-factor authentication (2FA) is an (optional) extra layer of security required to access the new Server Manager.
```

Author: "Optional" could lead to confusion

First, users will enter user name and password, then they will be required to provide a temporary verification code
generated by an application running on their smartphone.

2FA is disabled by default. Each user can enable it by accessing the :guilabel:`Two-factor authentication` section

> under :guilabel:`Settings` page, then follow these steps:

Reviewer 2: ```suggestion
under :guilabel:`Settings` page, then following these steps:
```

1. download and install the preferred 2FA application inside the smartphone
2. scan the QR code with the 2FA application
3. generate a new code and copy it inside :guilabel:`Verification code` field, than click :button:`Check code`
4. if the verification code is correct, click on the :button:`Save` button

Recovery codes
--------------

> Recovery codes can be used instead of temporary codes if the user can't access the 2FA application on the smartphone.

Reviewer 2: ```suggestion
Recovery codes can be used instead of temporary codes if the user cannot access the 2FA application on the smartphone.
```

> Recovery codes can be used instead of temporary codes if the user can't access the 2FA application on his own smartphone.

Reviewer: Recovery codes can be used instead of temporary codes if the user can't access the 2FA application on the smartphone.

Each recovery code is a one-time password and can be used only once.

To generate new recovery codes, disable 2FA, then re-enable it by registering the application again following the above steps.

Smartphone applications
-----------------------

There are several commercial and open source 2FA applications:

Available for both Android and iOS:

- `FreeOTP <https://freeotp.github.io/>`_: available for both Android and iOS
- `Authenticator <https://mattrubin.me/authenticator/>`_: available on iOS only
- `andOTP <https://github.com/andOTP/andOTP>`_: available for both Android and iOS https://github.com/andOTP/andOTP

Emergency recovery
------------------

> In case of emergeny, 2FA can be disabled accessing the server from a pysical console like a keyboard and a monitor,

Reviewer 3: pysical -> physical

Author: Thanks, I've forgot to run the spell checker!

> In case of emergeny, 2FA can be disabled accessing the server from a pysical console like a keyboard and a monitor,

Reviewer 3: emergeny -> emergency

a serial cable or a VNC-like connection for virtual machines:

1. access the system with user name and password
2. execute: ::
     rm -f ~/.2fa.secret

>      signal-event otp-save

Reviewer 2: Does it need sudo for normal users?

Author: no
